submissions/BlueLamantine/memory-pair-game/js/index.js

```diff
@@ -0,0 +1,229 @@
+      this.playAudio();
+    });
+  }
+}
+
+class Render {
+  constructor(areas) {
+    this.cardsArray = [
+      {
+        name: 'pawtter',
+        path: 'img/cats/pawtter.png',
+      },
+      {
+        name: 'hairmeowne',
+        path: 'img/cats/hairmeowne.png',
+      },
+      {
+        name: 'ron',
+        path: 'img/cats/ron.png',
+      },
+      {
+        name: 'lord',
+        path: 'img/cats/lord.png',
+      },
+      {
+        name: 'sevepuss',
+        path: 'img/cats/sevepuss.png',
+      },
+      {
+        name: 'dumpurrdore',
+        path: 'img/cats/dumpurrdore.png',
+      },
+    ];
+    this.areas = areas;
+    this.pawsInBlock = 10;
+    this.cardsInBlock = 3;
+  }
+
+  shuffleCards() {
+    const shuffleArray = this.cardsArray.concat(this.cardsArray).sort(() => {
+      return 0.5 - Math.random();
+    });
+    const cardsBack = document.querySelectorAll('.back');
+    shuffleArray.forEach((data, cardIndex) => {
+      cardsBack[cardIndex].innerHTML = `
+      <img name="${data.name}" src="${data.path}" alt="cat" class="card-img">
+      `;
+    });
+  }
+
+  getPawPrints() {
+    let template = ``;
+    for (let i = 0; i < this.pawsInBlock; i++) {
+      template += `<i class="fa fa-paw paw"></i>`;
+    }
+    return template;
+  }
+
+  renderBlock() {
+    let template = ``;
+    for (let i = 0; i < this.cardsInBlock; i++) {
+      template += `
+      <div class="wrap">
+        <div class="paws">
+          ${this.getPawPrints()}
+        </div>
+        <div class="card">
+          <div class="face front"></div>
+          <div class="face back"></div>
+        </div>
+      </div>`;
+    }
+    return template;
+  }
+
+  renderCards() {
+    this.areas.forEach(block => {
+      block.innerHTML = this.renderBlock();
+    });
+    this.shuffleCards();
+  }
+}
+
+class Game {
+  constructor() {
+    this.cards = document.querySelectorAll('.card');
+    this.openCardsCount = 0;
+    this.movesCount = 0;
+    this.maxOpenCards = 2;
+    this.animationDelay = {
+      flip: 300,
+      disappear: 520,
+    };
+    this.cardsNum = 12;
+    this.currentTime = '';
+  }
+
+  getDisappear(card) {
+    card.closest('.wrap').classList.add('hidden');
+  }
+
+  getFlip(card) {
+    card.closest('.card').querySelector('.front').classList.toggle('open');
+    card.closest('.card').querySelector('.back').classList.toggle('open');
+  }
+
+  gameOver () {
+    document.querySelector('#timer').classList.add('hidden');
+    document.querySelector('#moves-counter').classList.add('hidden');
+    const gameOver = document.createElement('div');
+    gameOver.classList.add('game-over-text');
+    gameOver.innerHTML =
+      `Congratulations!!! You broke the spell! ` +
+      `With ${this.movesCount} moves and your time is ${this.currentTime}.`;
+    document.querySelector('#map').appendChild(gameOver);
+  }
+
+  getMatching() {
+    let openedCards = Array.from(document.querySelectorAll('.front.open'));
```

Comment:
use const for this variable, please. you didn't any of the reassignment of it

Reply:
ok